sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -273,7 +273,9 @@ private[hive] class HiveClientImpl(
     if (clientLoader.cachedHive != null) {
       clientLoader.cachedHive.asInstanceOf[Hive]
     } else {
-      val c = Hive.get(conf)
+      // don't register all Hive permanent functions in Hive's FunctionRegistry since Spark loads
+      // them through direct HMS API calls
+      val c = Hive.getWithFastCheck(conf, false)
```

Comment:
So, are the previous errors irrecoverable?

Reply:
`Spark 3.x can't talk to HMS 1.2.x` sounds different from `Spark 3.x complains with Error`. For my understanding, I'm wondering if this PR only suppresses the error messages or fixes the real connectivity issues.

Reply:
The `Hive.getWithFastCheck` passes `false` to `doRegisterAllFns` in the PR description so it avoids calling `get_all_functions` from the remote HMS and fixes the compatibility issue.

Reply:
Yes, the question was the previous `get_all_functions` failure blocked the rest of Spark operations or not. If it blocks all further Spark's operations, it should have a `BUG` type.

Reply:
Got you. Yes it is unrecoverable. I'll update the JIRA to `BUG`.

Reply:
Thanks.

Reply:
Verify the failed tests with the following changes?
```scala
import java.lang.{Boolean => JBoolean}
val hiveClass = Utils.classForName("org.apache.hadoop.hive.ql.metadata.Hive")
val getInternalMethod = hiveClass.getDeclaredMethod("getInternal", classOf[HiveConf], classOf[Boolean], classOf[Boolean], classOf[Boolean])
getInternalMethod.setAccessible(true)
val c = getInternalMethod.invoke(hiveClass, conf, JBoolean.FALSE, JBoolean.FALSE, JBoolean.FALSE).asInstanceOf[Hive]
clientLoader.cachedHive = c
```